bot.ml

```diff
@@ -786,15 +786,181 @@ let callback _conn req body =
+                                   comment_info.author reviews_info.baseRef)
+                              ~id:pr.id
+                          else
+                            match reviews_info.review_decision with
+                            | NONE | REVIEW_REQUIRED ->
+                                GitHub_mutations.post_comment ~bot_info
+                                  ~message:
+                                    (f
+                                       "@%s: This PR cannot be merged because \
+                                        it hasn't been approved yet."
+                                       comment_info.author)
+                                  ~id:pr.id
+                            | CHANGES_REQUESTED ->
+                                GitHub_mutations.post_comment ~bot_info
+                                  ~message:
+                                    (f
+                                       "@%s: This PR cannot be merged because \
+                                        some changes are requested."
+                                       comment_info.author)
+                                  ~id:pr.id
+                            | APPROVED -> (
+                                if
+                                  not
+                                    (List.exists comment_info.issue.assignees
+                                       ~f:(fun login ->
+                                         String.equal login comment_info.author))
+                                then
+                                  GitHub_mutations.post_comment ~bot_info
+                                    ~message:
+                                      (f
+                                         "@%s: You can't merge the PR because \
+                                          you're not among the assignees."
+                                         comment_info.author)
+                                    ~id:pr.id
+                                else
+                                  GitHub_queries.get_team_membership ~bot_info
+                                    ~org:"coq" ~team:"pushers"
+                                    ~user:comment_info.author
+                                  >>= function
+                                  | Ok _ -> (
+                                      GitHub_mutations.merge_pull_request
+                                        ~bot_info ~pr_id:pr.id
+                                        ~commit_headline:
+                                          (f "Merge PR #%d: %s" pr.issue.number
+                                             comment_info.issue.title)
+                                        ~commit_body:
+                                          ( List.fold_left
+                                              reviews_info.approved_reviews
+                                              ~init:"Reviewed-by:\n"
+                                              ~f:(fun s r -> s ^ f "- %s\n" r)
```

Comment:
Something else which I've reviewed too fast in the past: we actually repeat `Reviewed-by:` and `Ack-by:` on each line (this is a `git` convention). Example: https://github.com/coq/coq/commit/55f4095fe3c366a9f310584a55e2dc0605e5409c